packages/framework/presence/src/presence.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ConnectedClientId } from "./baseTypes.js";
+import type { NotificationsManager } from "./notificationsManager.js";
+import type {
+	PresenceStates,
+	PresenceWorkspaceAddress,
+	PresenceStatesSchema,
+} from "./types.js";
+
+import type { ISubscribable } from "@fluid-experimental/presence/internal/events";
+import type { InternalTypes } from "@fluid-experimental/presence/internal/exposedInternalTypes";
+
+/**
+ * A client within a Fluid session (period of container connectivity to service).
+ *
+ * @sealed
+ * @alpha
+ */
+export interface ISessionClient {
+	/**
+	 * Get current client connection id.
+	 *
+	 * @returns Current client connection id.
+	 *
+	 * @remarks
+	 * Connection id will change on reconnection.
+	 */
+	currentClientId(): ConnectedClientId;
+}
+
+/**
+ * @sealed
+ * @alpha
+ */
+export interface PresenceEvents {
+	/**
+	 * Raised when new client joins session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeJoined: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when client appears disconnected from session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeDisconnected: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when a workspace is activated within the session.
+	 */
+	workspaceActivated: (
+		workspaceAddress: PresenceWorkspaceAddress,
+		type: "States" | "Notifications" | "Unknown",
+	) => void;
+}
+
+/**
+ * Presence represents known clients within a session and their custom states and notifications.
+ *
+ * @sealed
+ * @alpha
+ */
+export interface IPresence {
+	/**
+	 * Events for Notifications manager.
+	 */
+	readonly events: ISubscribable<PresenceEvents>;
+
+	/**
+	 * Get all attendees in the session.
+	 *
+	 * @remarks
+	 * Attendee states are dynamic and will change as clients join and leave
+	 * the session.
+	 */
+	getAttendees(): ReadonlySet<ISessionClient>;
```

Comment:
I was picking something distinct from the others that exist because this one will be a little different. Clients (`IClient`/unbranded strings that are `ConnectedClientId`) today are really client connections that come and go. Members (service audience) are user focused. Here we are going for a stable entity for a client independent of its connections/reconnections. In the end we should be able to get from anyone of those to the other ones. I agree it is complicated - each one makes more sense in the space they are in.
We've got work to do.